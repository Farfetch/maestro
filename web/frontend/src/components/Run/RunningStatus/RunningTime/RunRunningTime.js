import { Space, Statistic, Typography } from "antd";
import moment from "moment";

import Timer from "../../../Timer";

const RunRunningTime = ({ startedAt, finishedAt }) => {
  const title = "Running time:";

  return !finishedAt ? (
    <Timer startTime={startedAt}>
      {(time, loading) => (
        <Space>
          <Typography.Text type="secondary">{title}</Typography.Text>
          <Statistic title={""} value={time} loading={loading} />
        </Space>
      )}
    </Timer>
  ) : (
    <Space>
      <Typography.Text type="secondary">{title}</Typography.Text>
      <Statistic
        title={""}
        value={moment.utc(finishedAt.diff(startedAt)).format("HH:mm:ss")}
      />
    </Space>
  );
};

export default RunRunningTime;
