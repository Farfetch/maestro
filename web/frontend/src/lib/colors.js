import {
  blue,
  cyan,
  geekblue,
  gold,
  green,
  grey,
  lime,
  magenta,
  orange,
  purple,
  red,
  volcano,
  yellow
} from "@ant-design/colors";

export const colors = {
  red,
  volcano,
  orange,
  gold,
  yellow,
  lime,
  green,
  cyan,
  blue,
  geekblue,
  purple,
  magenta,
  grey
};

export const defaultColorsList = [
  blue,
  geekblue,
  purple,
  magenta,
  cyan,
  red,
  volcano,
  orange,
  gold,
  yellow,
  lime,
  green
];

export const getNextColor = (colorIndex, colorsList = defaultColorsList) => {
  const color = colorsList[colorIndex];
  let nextColorIndex = colorIndex;

  if (colorsList[colorIndex + 1]) {
    nextColorIndex += 1;
  } else {
    nextColorIndex = 0;
  }

  return { colorIndex: nextColorIndex, color };
};
