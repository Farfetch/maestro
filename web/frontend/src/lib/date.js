import moment from "moment";

const defaultDatetimeFormat = "YYYY-MM-DD HH:mm:ss";

const hourMinuteFormat = "HH:mm";

export const toUtcString = (localDatetime) =>
  localDatetime.utc().format(defaultDatetimeFormat);

export const toLocalDate = (utcDatetimeStr) =>
  moment.utc(utcDatetimeStr).local();

export const toUtcHourMinute = (localDatetime) =>
  localDatetime.utc().format(hourMinuteFormat);

export const toLocalHourMinute = (utcHourMinute) =>
  moment.utc(utcHourMinute, hourMinuteFormat).local();
