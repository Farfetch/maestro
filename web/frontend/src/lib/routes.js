import { maestroApiUrl } from "../config";

export const runPlanDownloadUrl = (runPlanId) =>
  `${maestroApiUrl}/api/run_plan/${runPlanId}/download`;

export const customDataDownloadUrl = (customDataId) =>
  `${maestroApiUrl}/api/custom_data/${customDataId}/download`;

export const runMetricsDownloadUrl = (runId) =>
  `${maestroApiUrl}/api/run_metrics/${runId}/download`;

export const runLogDownloadUrl = (runId) =>
  `${maestroApiUrl}/api/run_log/${runId}/download`;

export const testSingleUrl = (runConfigurationId) =>
  `/test/${runConfigurationId}`;

export const testNewUrl = "/tests/new";

export const agentLogsUrl = (runConfigurationId) =>
  `/agent/${runConfigurationId}/logs`;

export const runSingleUrl = (runId) => `/run/${runId}`;

export const historyUrl = "/history";

export const logoutUrl = `${maestroApiUrl}/logout`;
export const workspacesUrl = "/workspaces";
export const usersUrl = "/users";
