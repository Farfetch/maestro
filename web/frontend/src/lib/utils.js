export const avg = (value, itemsCount) =>
  itemsCount === 0 ? 0 : value / itemsCount;

export const calculateErrorRate = (successCount, totalCount) => {
  const errorRate = parseFloat((1 - successCount / totalCount) * 100).toFixed(
    2
  );

  return errorRate === "NaN" ? 0 : errorRate;
};
