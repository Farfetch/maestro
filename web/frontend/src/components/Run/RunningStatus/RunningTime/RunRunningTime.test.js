import { render } from "@testing-library/react";
import mockDate from "mockdate";
import React from "react";

import { toLocalDate } from "../../../../lib/date";
import RunRunningTime from "./RunRunningTime";

describe("components/Run/RunningStatus/RunRunningTime", () => {
  test(`should render Statistic component`, async () => {
    const startedAt = toLocalDate("2021-04-14 12:29:20");
    const finishedAt = toLocalDate("2021-04-14 14:29:20");

    const rendered = render(
      <RunRunningTime startedAt={startedAt} finishedAt={finishedAt} />
    );

    const renderedContainer = rendered.container;
    expect(renderedContainer.outerHTML).toMatchSnapshot();
  });

  test(`should render Timer component`, async () => {
    const startedAt = toLocalDate("2021-04-14 12:29:20");
    mockDate.set(toLocalDate("2021-04-14 12:31:20"));

    const rendered = render(
      <RunRunningTime startedAt={startedAt} finishedAt={false} />
    );

    const renderedContainer = rendered.container;
    expect(renderedContainer.outerHTML).toMatchSnapshot();

    mockDate.reset();
  });
});
