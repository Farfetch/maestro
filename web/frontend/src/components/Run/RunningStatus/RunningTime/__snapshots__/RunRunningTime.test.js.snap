// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Run/RunningStatus/RunRunningTime should render Statistic component 1`] = `"<div><div class=\\"ant-space css-dev-only-do-not-override-nnuwmp ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><span class=\\"ant-typography ant-typography-secondary css-dev-only-do-not-override-nnuwmp\\">Running time:</span></div><div class=\\"ant-space-item\\"><div class=\\"ant-statistic css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\">02:00:00</span></div></div></div></div></div>"`;

exports[`components/Run/RunningStatus/RunRunningTime should render Timer component 1`] = `"<div><div class=\\"ant-space css-dev-only-do-not-override-nnuwmp ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><span class=\\"ant-typography ant-typography-secondary css-dev-only-do-not-override-nnuwmp\\">Running time:</span></div><div class=\\"ant-space-item\\"><div class=\\"ant-statistic css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\">00:02:00</span></div></div></div></div></div>"`;
